src/engine/imgui_layer.cpp

```diff
@@ -222,16 +194,38 @@ void MapConfigEditor::Display() {
     manifest_.ambient_colour.g = plFloatToByte(rgb[1]);
     manifest_.ambient_colour.b = plFloatToByte(rgb[2]);
 
+    ImGui::Separator();
+
+    const char *temperatures[]={ "hot", "cold" };
+
+    //ImGui::ListBox("Temperature", temperatures, , );
+
     ImGui::End();
 
-    if(open_window_->GetStatus()) {
-        open_window_->Display();
+    if(save_dialog) {
+        ImGui::SetNextWindowSize(ImVec2(256, 128), ImGuiCond_Once);
+        ImGui::Begin(dname("Map Config Editor / Save"), &save_dialog, default_flags);
+        ImGui::InputText("Name", filename_buffer, sizeof(filename_buffer));
+        ImGui::Separator();
+        if(ImGui::Button("Save")) {
+            SaveManifest(std::string(GetFullCampaignPath()) + "/maps/" + filename_buffer + ".map");
+        }
+        ImGui::SameLine();
+        if(ImGui::Button("Cancel")) {
+            save_dialog = false;
+        }
+        ImGui::End();
     }
 }
 
-void MapConfigEditor::OpenManifest(const std::string &name) {
+void MapConfigEditor::OpenManifest(const std::string &path) {
+    std::string name = plGetFileName(path.c_str());
+    name.erase(name.find(".map"));
     manifest_ = *MapManager::GetInstance()->GetManifest(name);
+
     SyncManifest();
+
+    strncpy(filename_buffer, path.c_str(), sizeof(filename_buffer));
```

Comment:
This won't terminate filename_buffer if path is too long. Would these class variables be best as C++ strings?

Reply:
Will fix it, but ImGui writes into that so it can't be std string.

Reply:
Can't as ImGui doesn't support filling std string.